crates/matrix-sdk-ui/src/timeline/inner/mod.rs

```diff
@@ -844,11 +780,42 @@ impl<P: RoomDataProvider> TimelineInner<P> {
             txn.commit();
 
             debug!("Discarded local echo");
-            true
-        } else {
-            debug!("Can't find local echo to discard");
-            false
+            return true;
+        }
+
+        // Look if this was a local reaction echo.
+        if let Some(full_key) =
+            state.meta.reactions.map.remove(&TimelineEventItemId::TransactionId(txn_id.to_owned()))
+        {
+            let item = match &full_key.item {
+                TimelineEventItemId::TransactionId(_) => {
+                    // TODO(bnjbvr): reactions on local echoes
+                    unimplemented!("no reactions on local echoes yet")
```

Comment:
Prefer a `error!` instead of a panic?

Reply:
This code path is unreachable, and transient until we get https://github.com/matrix-org/matrix-rust-sdk/pull/3749, but I'll put an `error!` just in case.